tests/device_tests/cardano/test_sign_tx.py

```diff
@@ -39,27 +39,30 @@ def test_cardano_sign_tx(client, parameters, result):
     certificates = [cardano.create_certificate(c) for c in parameters["certificates"]]
     withdrawals = [cardano.create_withdrawal(w) for w in parameters["withdrawals"]]
 
-    expected_responses = [messages.PassphraseRequest()]
-    expected_responses += [
-        messages.ButtonRequest(code=messages.ButtonRequestType.Other)
-        for i in range(len(parameters["input_flow"]))
+    input_flow = parameters.get("input_flow", ())
+
+    expected_responses = [
+        messages.ButtonRequest(),
+        messages.Success(),
+        messages.Features(),
+        messages.PassphraseRequest(),
     ]
+    expected_responses += _to_expected_responses(input_flow)
     expected_responses.append(messages.CardanoSignedTx())
 
-    def input_flow():
-        for sequence in parameters["input_flow"]:
-            yield
-            for action in sequence:
-                if action == "SWIPE":
-                    client.debug.swipe_up()
-                elif action == "YES":
-                    client.debug.press_yes()
-                else:
-                    raise ValueError("Invalid input action")
-
     with client:
         client.set_expected_responses(expected_responses)
-        client.set_input_flow(input_flow)
+        client.set_input_flow(_to_device_actions(client, [["YES"]] + input_flow))
+
+        if parameters.get("security_checks", "") == "prompt":
```

Comment:
move this outside the `with client` block to avoid messing up the expected responses / input flows

Reply:
Oh, nice, thanks. Done https://github.com/trezor/trezor-firmware/pull/1423/commits/b61d2fa5cfd83aa04bdde76084a7b0cd0c25512c.